test/test.cpp

```diff
@@ -462,7 +462,114 @@ int do_test_pow(uint64_t aStartPos, int aLabelsCount, unsigned aDiff, unsigned a
 						printf("%u hashes, %u h/s, solution at %u\n", (uint32_t)hashes_computed, (uint32_t)hashes_per_sec, (uint32_t)idx_solution);
 						if (-1 != cpu_id) {
 							uint8_t hash[32];
-							scryptPositions(cpu_id, s_id, idx_solution, idx_solution, 256, s_salt, SPACEMESH_API_COMPUTE_LEAFS, hash, 512, 1, 1, NULL, NULL, &hashes_computed, &hashes_per_sec);
+							scryptPositions(cpu_id, s_id, idx_solution, idx_solution, 256, s_salt, SPACEMESH_API_COMPUTE_POW, hash, 512, 1, 1, NULL, NULL, &hashes_computed, &hashes_per_sec);
+							printf("id: ");
+							print_hex32(s_id);
+							printf("\n");
+							printf("salt: ");
+							print_hex32(s_salt);
+							printf("\n");
+							printf("D: ");
+							print_hex32(D);
+							printf("\n");
+							printf("H: ");
+							print_hex32(hash);
+							printf("\n");
+						}
+						break;
+					case SPACEMESH_API_ERROR_NONE:
+						printf("%u hashes, %u h/s, solution not found\n", (uint32_t)hashes_computed, (uint32_t)hashes_per_sec);
+						break;
+					default:
+						printf("error %d, %u hashes, %u h/s\n", status, (uint32_t)hashes_computed, (uint32_t)hashes_per_sec);
+					}
+					if (aSolutionIdx != 0xffffffffffffffffull) {
+						if (idx_solution != aSolutionIdx) {
+							printf("WRONG solution index %ull, expected %ull\n", idx_solution, aSolutionIdx);
+							return 1;
+						}
+					}
+				}
+			}
+
+			return 0;
+		}
+	}
+
+	return 1;
+}
+
+/* test PoW */
```

Comment:
This comment doesn't have much value and it's the same as the test above. Could you elaborate more on what this test is about?

Reply:
updated